packages/core/src/render3/styling/style_differ.ts

```diff
@@ -60,24 +59,8 @@ export function computeStyleChanges(oldValue: string, newValue: string): StyleCh
  * @param isNewValue `true` if parsing new value (effects how values get added to `changes`)
  */
 export function parseKeyValue(text: string, changes: StyleChangesMap, isNewValue: boolean): void {
-  const end = text.length;
-  let start = 0;
-  while (start < end) {
-    const keyStart = consumeWhitespace(text, start, end);
-    const keyEnd = consumeStyleKey(text, keyStart, end);
-    if (keyEnd === keyStart) {
-      // we reached an end so just quit
-      break;
-    }
-    const valueStart = consumeSeparator(text, keyEnd, end, CharCode.COLON);
-    const valueEnd = consumeStyleValue(text, valueStart, end);
-    if (ngDevMode && valueStart === valueEnd) {
-      throw malformedStyleError(text, valueStart);
-    }
-    start = consumeSeparator(text, valueEnd, end, CharCode.SEMI_COLON);
-    const key = text.substring(keyStart, keyEnd);
-    const value = text.substring(valueStart, valueEnd);
-    processStyleKeyValue(changes, key, value, isNewValue);
+  for (let i = parseStyle(text); (i = parseStyleNext(text, i)) >= 0;) {
```

Comment:
This format is a bit hard to read because the condition and the iterator are combined.  If we add a `parseStyleNext(...)` invocation to the end of the `parseStyle` method, we can make the iteration look more idiomatic/readable:

```suggestion
  for (let i = parseStyle(text); i >= 0; i = parseStyleNext(text, i)) {
```